osquery/tables/system/darwin/xprotect.cpp

```diff
@@ -26,11 +26,32 @@ namespace fs = boost::filesystem;
 namespace osquery {
 namespace tables {
 
-/// Path to XProtect.meta.plist and XProtect.plist
-const std::string kXProtectPath =
+/// Path to XProtect.meta.plist and XProtect.plist, before macOS 11
+const std::string kXProtectPathPreBigSur =
     "/System/Library/CoreServices/"
     "CoreTypes.bundle/Contents/Resources/";
 
+/// the path to the XProtect.meta.plist and XProtect.plist changed in macOS 11
+const std::string kXProtectPathPostBigSur =
+    "/Library/Apple/System/Library/"
+    "CoreServices/XProtect.bundle/Contents/Resources";
+
+Status getExpectedXProtectDir(std::string& path) {
```

Comment:
I'd suggest less logic here, and using an array of search paths. Then take the first file that exists.

Reply:
I can do that, but the old directory _does_ exist on Big Sur, so we'll need to check for the existence of the file in question (`XProtect.plist` and `XProtect.meta.plist`) in for each `genXProtect{Meta,Entries}` method

Reply:
Sure. The full path. I'd probably Fitch this function too. There are likely some other examples.